BTCPayServer/Services/Altcoins/Ethereum/Services/EthereumService.cs

```
@@ -0,0 +1,313 @@
+            {
+                try
+                {
+                    var settings = await _settingsRepository.GetSettingAsync<EthereumLikeConfiguration>(
+                        EthereumLikeConfiguration.SettingsKey(chainId));
+                    if (settings is null || string.IsNullOrEmpty(settings.Web3ProviderUrl))
+                    {
+                        var val = _configuration.GetValue<string>($"chain{chainId}_web3", null);
+                        var valUser = _configuration.GetValue<string>($"chain{chainId}_web3_user", null);
+                        var valPass = _configuration.GetValue<string>($"chain{chainId}_web3_password", null);
+                        if (val != null && First)
+                        {
+                            Logs.PayServer.LogInformation($"Setting eth chain {chainId} web3 to {val}");
+                            settings ??= new EthereumLikeConfiguration()
+                            {
+                                ChainId = chainId,
+                                Web3ProviderUrl = val,
+                                Web3ProviderPassword = valPass,
+                                Web3ProviderUsername = valUser
+                            };
+                            await _settingsRepository.UpdateSetting(settings, EthereumLikeConfiguration.SettingsKey(chainId));
+                        }
+                    }
+                    var currentlyRunning = _chainHostedServices.ContainsKey(chainId);
+                    var valid = await EthereumConfigController.CheckValid(_httpClientFactory, _btcPayNetworkProvider.NetworkType, settings?.InvoiceId);
+                    if (!currentlyRunning  || (currentlyRunning && !valid))
+                    {
+                        await HandleChainWatcher(settings, valid, cancellationToken);
+                    }
+                }
+                catch (Exception)
+                {
+                    // ignored
```

swallowing exceptions liberally can lead to bugs that are very hard to catch; shouldn't this exception be logged somewhere at least? and put the reason of why this can be ignored, in the comment (sorry I'm no maintainer of BTCPayServer but as I was very interested in ether support, I might start looking at this closer now, if not start to contribute)

---

This loop attempt to start listen to all support eth chain. if 1 crash, it not matter as it retry later (as it possible config issue) while need to move on to start others.